pkg/reconciler/workload/apiexport/workload_apiexport_reconcile.go

```diff
@@ -198,38 +185,20 @@ func (r *schemaReconciler) reconcile(ctx context.Context, export *apisv1alpha1.A
 	}
 
 	// delete schemas that are no longer needed
-	allSchemas, err := r.listAPIResourceSchemas(clusterName)
-	if err != nil {
-		return reconcileStatusStop, err
-	}
-	for _, schema := range allSchemas {
-		if !referencedSchemaNames[schema.Name] && metav1.IsControlledBy(schema, export) {
-			logging.WithObject(logger, schema).V(2).Info("deleting schema of APIExport")
-			if err := r.deleteAPIResourceSchema(ctx, clusterName.Path(), schema.Name); err != nil && !apierrors.IsNotFound(err) {
-				return reconcileStatusStop, err
+	for schemaName := range outDatedSchemaNames {
+		if _, err := r.getAPIResourceSchema(ctx, clusterName, schemaName); err != nil {
```

Comment:
Why do we need to do the get first ? We don't use the result.

Reply:
we do not need this. It was only to avoid an unnecessary delete. But we can always delete it.